src/handlers/motions/motionCreated/handlers/setUserRoles.ts

```diff
@@ -32,6 +41,38 @@ export const handleSetUserRolesMotion = async (
     colonyRolesDatabaseId,
   );
 
+  const {
+    id: existingColonyRoleId,
+    latestBlock: existingColonyRoleLatestBlock = 0,
+    /*
+     * We need to extract __typename since the `existingRoles` object will get
+     * Passed down to another mutation and typenames will clash
+     */
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    __typename,
+    ...existingRoles
+  } = (
+    await query<GetColonyRoleQuery, GetColonyRoleQueryVariables>(
+      GetColonyRoleDocument,
+      { id: colonyRolesDatabaseId },
+    )
+  )?.data?.getColonyRole ?? {};
+
+  // only create a historic role if it's not the first assignment of roles
+  if (existingColonyRoleId && blockNumber > existingColonyRoleLatestBlock) {
+    await createColonyHistoricRoleDatabaseEntry(
+      colonyAddress,
+      domainId.toNumber(),
+      userAddress,
+      blockNumber,
+      {
+        ...existingRoles,
+        ...roles,
+      },
+      isMultiSig,
+    );
+  }
```

Comment:
Given we use the same piece of code in both `src/handlers/motions/motionCreated/handlers/setUserRoles.ts` and `src/handlers/multiSig/multiSigCreated/handlers/setUserRoles.ts`, could we extract it in a helper function?